q/compare.go

```diff
@@ -48,6 +48,18 @@ func compare(a, b interface{}, tok token.Token) bool {
 
 			return constant.Compare(constant.MakeFloat64(vala.Float()), tok, constant.MakeFloat64(bla))
 		}
+	case ak == reflect.Struct:
```

Comment:
The problem now is that if we try to compare a struct with another struct using the `EQ` token it won't work because structs are no longer catched by the last case `case tok == token.EQL:`
I think we should keep a `case` by type and add a copy of the last `case` to an `if` statement after line 61